File: fixtures/scale/dart/large/lib/ciphers/foursquare/foursquare_key.dart
import '../playfair/playfair_key.dart';

class FourSquareKey {
  final String upperWord;
  final String lowerWord;

  FourSquareKey(this.upperWord, this.lowerWord);

  String upperSquare() => PlayfairKey(upperWord).square();

  String lowerSquare() => PlayfairKey(lowerWord).square();

  factory FourSquareKey.defaultKey() => FourSquareKey('WINTER', 'SUMMER');
}

File: fixtures/scale/kotlin/large/src/cryptobench/ciphers/foursquare/FourSquareKey.kt
package cryptobench.ciphers.foursquare

import cryptobench.ciphers.playfair.PlayfairKey

data class FourSquareKey(val upperWord: String, val lowerWord: String) {

    fun upperSquare(): String = PlayfairKey(upperWord).square()

    fun lowerSquare(): String = PlayfairKey(lowerWord).square()

    companion object {
        fun default(): FourSquareKey = FourSquareKey("WINTER", "SUMMER")
    }
}
